testsuite/tests/compaction/test_quick_compact.ml

```diff
@@ -0,0 +1,41 @@
+let keep_alive = ref []
+
+let generate_garbage () =
+  keep_alive := [];
+  (* start with a very similar heap state *)
+  begin
+    Gc.full_major ();
+    Gc.compact ();
```

Comment:
Unless I am missing something, calling `full_major` before `compact` is now redundant.